File: src/scala/MinMaxIterativo.scala

def minMax(data: List[Int]): (Int, Int) = {
  var min: Int = data(0)
  var max: Int = data(0)
  for (value <- data) {
    if (value < min) {
      min = value
    }
    if (value > max) {
      max = value
    }
  }
  (min, max)
}

object Main extends App {
  val data: List[Int] = List(4, 6, 2, 9, 3, 8, 1, 7, 5)
  val (min, max) = minMax(data)
  println(s"$data\nMin: $min\nMax: $max")
}

File: src/java/MinMaxIterativo.java
public class MinMaxIterativo {
    public static void main(String[] args){
        int arr[] = {1, 5, -20, 6, 15};
        int min = arr[0];
        int max = arr[0];

        for (int i:arr) {
            if(i < min) min = i;
            else if(i > max) max = i;
        }
        System.out.println("Max: " + max + "\nMin: " + min);
    }
}
